lib/api/apiUtils/object/versioning.js

```diff
@@ -401,6 +403,18 @@ function overwritingVersioning(objMD, metadataStoreParams) {
     metadataStoreParams.creationTime = objMD['creation-time'];
     metadataStoreParams.lastModifiedDate = objMD['last-modified'];
     metadataStoreParams.updateMicroVersionId = true;
+
+    // update restore
+    const days = objMD.archive?.restoreRequestedDays;
```

Comment:
Yes, earlier in the API process we check that `objMD.archive.restoreRequestedDays` is defined.
Since `overwritingVersioning()` could be reused, I used this question mark as a safeguard.